plugins/Hosting/js/sagas/saga.js

```diff
@@ -204,6 +206,33 @@ function *requestDefaultSettingsSaga() {
 	}
 }
 
+// getHostStatus returns a promise that resolves with `true` if the provided
+// netaddress is connectable, or `false` if it is not connectable. The input
+// 'hostAddr' must be an address in the URL format, eg:
+// tcp://123.456.789.0:9982
+const getHostStatus = (hostAddr) => new Promise((resolve) => {
+	const hostUrl = url.parse(hostAddr)
+	const socket = net.connect(hostUrl.port, hostUrl.hostname)
+	socket.setTimeout(15e3) // 15 second host connectability timeout
```

Comment:
Less than 10 minutes to get your first ping.

Interestingly, seems that hosts can get pings even if their ports are not open. Presumably from their own hostdb? Not sure